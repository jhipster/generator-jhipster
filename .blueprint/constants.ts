import { join } from 'path';
import { getPackageRoot } from '../lib/index.js';

const packageRoot = getPackageRoot();
export const defaultSamplesFolder = join(packageRoot, '../jhipster-samples');
export const testIntegrationFolder = join(packageRoot, 'test-integration');
export const samplesFolder = join(testIntegrationFolder, 'samples');

export const jhipsterBin = join(packageRoot, 'bin/jhipster.cjs');

export const jdlSamplesFolder = join(testIntegrationFolder, 'jdl-samples');
export const dailyBuildsFolder = join(testIntegrationFolder, 'daily-builds');
export const jdlEntitiesSamplesFolder = join(samplesFolder, 'jdl-entities');

export const entitiesSamplesDir = join(samplesFolder, '.jhipster');
