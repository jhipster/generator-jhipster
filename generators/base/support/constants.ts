export const LOCAL_BLUEPRINT_PACKAGE_NAMESPACE = '@jhipster/generator-jhipster-local';
