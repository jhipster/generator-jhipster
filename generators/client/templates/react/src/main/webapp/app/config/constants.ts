const config = {
  version: process.env.version || '0.0.0'
};

export default config;

export const messages = {
  DATA_CREATE_SUCCESS_ALERT: 'Data created successfully!',
  DATA_DELETE_SUCCESS_ALERT: 'Data deleted successfully!',
  DATA_UPDATE_ERROR_ALERT: 'Data deleted successfully!'
};
