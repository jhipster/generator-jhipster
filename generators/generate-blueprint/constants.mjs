/**
 * Copyright 2013-2022 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import chalk from 'chalk';

import GENERATOR_LIST from '../generator-list.js';
import { BASE_PRIORITY_NAMES, ENTITY_PRIORITY_NAMES } from '../../lib/constants/priorities.mjs';

const prioritiesForSub = subGenerator => (subGenerator.startsWith('entit') ? ENTITY_PRIORITY_NAMES : BASE_PRIORITY_NAMES);

export const GENERATORS = 'generators';
export const SUB_GENERATORS = 'subGenerators';
export const DYNAMIC = 'dynamic';
export const JS = 'js';

export const SBS = 'sbs';
export const COMMAND = 'command';
export const PRIORITIES = 'priorities';
export const ALL_GENERATORS = 'allGenerators';
export const WRITTEN = 'written';

/**
 * Options exposed to cli
 */
export const options = () => ({
  [SUB_GENERATORS]: {
    desc: 'Sub generators to generate',
    type: Array,
    scope: 'storage',
  },
  [DYNAMIC]: {
    desc: 'Genate dynamic generators (advanced)',
    type: Boolean,
    scope: 'storage',
  },
  [JS]: {
    desc: 'Use js extension',
    type: Boolean,
    scope: 'storage',
  },
  [ALL_GENERATORS]: {
    desc: 'Use js extension',
    type: Boolean,
    scope: 'generator',
  },
});

/**
 * Config that needs to be written to config
 */
export const requiredConfig = () => ({});

/**
 * Default config that will be used for templates
 */
export const defaultConfig = () => ({
  ...requiredConfig,
  [DYNAMIC]: false,
  [JS]: false,
  [SUB_GENERATORS]: [],
});

export const defaultSubGeneratorConfig = () => ({
  [SBS]: true,
  [COMMAND]: false,
  [WRITTEN]: false,
  [PRIORITIES]: [],
});

const allSubGeneratorConfig = subGenerator => ({
  [SBS]: true,
  [COMMAND]: false,
  [PRIORITIES]: prioritiesForSub(subGenerator),
});

export const allGeneratorsConfig = () => ({
  ...requiredConfig,
  [SUB_GENERATORS]: Object.values(GENERATOR_LIST),
  [DYNAMIC]: false,
  [JS]: false,
  [GENERATORS]: Object.fromEntries(
    Object.values(GENERATOR_LIST).map(subGenerator => {
      return [subGenerator, allSubGeneratorConfig(subGenerator)];
    })
  ),
});

export const prompts = () => {
  return [
    {
      type: 'checkbox',
      name: SUB_GENERATORS,
      message: 'Which sub-generators do you want to override?',
      choices: Object.values(GENERATOR_LIST),
      pageSize: 30,
      loop: false,
    },
    {
      type: 'confirm',
      name: 'cli',
      message: 'Add a cli?',
      default: true,
    },
  ];
};

export const subGeneratorPrompts = subGenerator => {
  return [
    {
      type: 'confirm',
      name: SBS,
      message: `Is ${chalk.yellow(subGenerator)} generator a side-by-side blueprint?`,
      default: true,
    },
    {
      type: 'confirm',
      name: COMMAND,
      message: `Is ${chalk.yellow(subGenerator)} generator a cli command?`,
      default: false,
    },
    {
      type: 'checkbox',
      name: PRIORITIES,
      message: `What task do you want do implement at ${chalk.yellow(subGenerator)} generator?`,
      choices: prioritiesForSub(subGenerator),
      pageSize: 30,
      default: answers => (answers.sbs ? [] : prioritiesForSub(subGenerator)),
      loop: false,
    },
  ];
};
