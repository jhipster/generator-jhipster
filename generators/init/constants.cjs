/**
 * Copyright 2013-2022 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const NODE_VERSION = '14.18.1';

const PRETTIER_DEFAULT_INDENT = 'prettierDefaultIndent';
const PRETTIER_DEFAULT_INDENT_DEFAULT_VALUE = 2;

const SKIP_COMMIT_HOOK = 'skipCommitHook';
const SKIP_COMMIT_HOOK_DESCRIPTION = 'Skip adding husky commit hooks';
const SKIP_COMMIT_HOOK_DEFAULT_VALUE = false;

module.exports = {
  NODE_VERSION,
  PRETTIER_DEFAULT_INDENT,
  PRETTIER_DEFAULT_INDENT_DEFAULT_VALUE,
  SKIP_COMMIT_HOOK,
  SKIP_COMMIT_HOOK_DESCRIPTION,
  SKIP_COMMIT_HOOK_DEFAULT_VALUE,
};
