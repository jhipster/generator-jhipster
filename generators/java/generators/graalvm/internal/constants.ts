// renovate: datasource=github-releases depName=graalvm-reachability-metadata oracle/graalvm-reachability-metadata
export const GRAALVM_REACHABILITY_METADATA = '0.3.14';
