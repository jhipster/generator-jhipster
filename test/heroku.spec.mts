import assert from 'yeoman-assert';
import helpers from 'yeoman-test';
import fse from 'fs-extra';
import sinon from 'sinon';
import ChildProcess from 'child_process';
import { jestExpect as expect } from 'mocha-expect-snapshot';

import constants from '../generators/generator-constants.cjs';
import { getTemplatePath, getGenerator } from './support/index.mjs';

const expectedFiles = {
  monolith: [
    'Procfile',
    `${constants.SERVER_MAIN_RES_DIR}/config/bootstrap-heroku.yml`,
    `${constants.SERVER_MAIN_RES_DIR}/config/application-heroku.yml`,
  ],
};

describe('JHipster Heroku Sub Generator', () => {
  const herokuAppName = 'jhipster-test';
  let stub;

  beforeEach(() => {
    stub = sinon.stub(ChildProcess, 'exec');
    stub.withArgs('heroku --version').yields(false);
    stub.withArgs('heroku plugins').yields(false, 'heroku-cli-deploy');
    stub.withArgs('git init').yields([false, '', '']);
  });
  afterEach(() => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (ChildProcess.exec as any).restore();
  });

  describe('microservice application', () => {
    describe('with JAR deployment', () => {
      let runResult;
      beforeEach(async () => {
        stub.withArgs(`heroku create ${herokuAppName}`).yields(false, '', '');
        stub.withArgs(`heroku addons:create jawsdb:kitefin --as DATABASE --app ${herokuAppName}`).yields(false, '', '');
        stub
          .withArgs(`heroku config:set JHIPSTER_REGISTRY_URL=https://admin:changeme@sushi.herokuapp.com --app ${herokuAppName}`)
          .yields(false, '', '')
          .returns({
            stdout: {
              // eslint-disable-next-line @typescript-eslint/no-empty-function
              on: () => {},
            },
          });
        runResult = await helpers
          .create(getGenerator('heroku'))
          .inTmpDir(dir => {
            fse.copySync(getTemplatePath('default-microservice/'), dir);
          })
          .withOptions({ skipBuild: true, reproducibleTests: true })
          .withPrompts({
            herokuAppName,
            herokuRegion: 'us',
            herokuDeployType: 'jar',
            herokuJHipsterRegistryApp: 'sushi',
            herokuJHipsterRegistryUsername: 'admin',
            herokuJHipsterRegistryPassword: 'changeme',
            herokuJavaVersion: '17',
            useOkta: false,
          })
          .run();
      });
      it('should match files snapshot', function () {
        expect(runResult.getSnapshot()).toMatchSnapshot();
      });
      it('creates expected files', () => {
        assert.fileContent('.yo-rc.json', '"herokuDeployType": "jar"');
      });
    });
  });

  describe('monolith application', () => {
    describe('with an unavailable app name', () => {
      const autogeneratedAppName = 'jhipster-new-name';
      let runResult;
      beforeEach(async () => {
        stub
          .withArgs(`heroku create ${herokuAppName}`)
          .yields(true, '', `Name ${herokuAppName} is already taken`)
          .returns({
            stdout: {
              // eslint-disable-next-line @typescript-eslint/no-empty-function
              on: () => {},
            },
          });
        stub.withArgs('heroku create ').yields(false, `https://${autogeneratedAppName}.herokuapp.com`);
        stub.withArgs(`heroku git:remote --app ${autogeneratedAppName}`).yields(false, `https://${autogeneratedAppName}.herokuapp.com`);
        stub.withArgs(`heroku addons:create jawsdb:kitefin --as DATABASE --app ${autogeneratedAppName}`).yields(false, '', '');
        runResult = await helpers
          .create(getGenerator('heroku'))
          .inTmpDir(dir => {
            fse.copySync(getTemplatePath('default'), dir);
          })
          .withOptions({ skipBuild: true, reproducibleTests: true })
          .withPrompts({
            herokuAppName,
            herokuRegion: 'us',
            herokuDeployType: 'jar',
            herokuForceName: 'No',
            herokuJavaVersion: '11',
            useOkta: false,
          })
          .run();
      });
      it('should match files snapshot', function () {
        expect(runResult.getSnapshot()).toMatchSnapshot();
      });
      it('creates expected monolith files', () => {
        assert.file(expectedFiles.monolith);
        assert.fileContent('.yo-rc.json', `"herokuAppName": "${autogeneratedAppName}"`);
      });
    });

    describe('with Git deployment', () => {
      let runResult;
      beforeEach(async () => {
        stub.withArgs(`heroku create ${herokuAppName}`).yields(false, '', '');
        stub.withArgs(`heroku addons:create jawsdb:kitefin --as DATABASE --app ${herokuAppName}`).yields(false, '', '');
        stub.withArgs('git add .').yields(false, '', '');
        stub.withArgs('git commit -m "Deploy to Heroku" --allow-empty').yields(false, '', '');
        stub.withArgs(`heroku config:set MAVEN_CUSTOM_OPTS="-Pprod,heroku -DskipTests" --app ${herokuAppName}`).yields(false, '', '');
        stub.withArgs(`heroku buildpacks:add heroku/java --app ${herokuAppName}`).yields(false, '', '');
        stub.withArgs('git push heroku HEAD:master').yields(false, '', '');
        runResult = await helpers
          .create(getGenerator('heroku'))
          .inTmpDir(dir => {
            fse.copySync(getTemplatePath('default'), dir);
          })
          .withPrompts({
            herokuAppName,
            herokuRegion: 'us',
            herokuDeployType: 'git',
            herokuJavaVersion: '11',
            useOkta: false,
          })
          .run();
      });
      it('should match files snapshot', function () {
        expect(runResult.getSnapshot()).toMatchSnapshot();
      });
      it('creates expected monolith files', () => {
        assert.file(expectedFiles.monolith);
        assert.fileContent('.yo-rc.json', '"herokuDeployType": "git"');
      });
    });

    describe('in the US', () => {
      let runResult;
      beforeEach(async () => {
        stub.withArgs(`heroku create ${herokuAppName}`).yields(false, '', '');
        stub.withArgs(`heroku addons:create jawsdb:kitefin --as DATABASE --app ${herokuAppName}`).yields(false, '', '');
        runResult = await helpers
          .create(getGenerator('heroku'))
          .inTmpDir(dir => {
            fse.copySync(getTemplatePath('default/'), dir);
          })
          .withOptions({ skipBuild: true, reproducibleTests: true })
          .withPrompts({
            herokuAppName,
            herokuRegion: 'us',
            herokuDeployType: 'jar',
            herokuJavaVersion: '11',
            useOkta: false,
          })
          .run();
      });
      it('should match files snapshot', function () {
        expect(runResult.getSnapshot()).toMatchSnapshot();
      });
      it('creates expected monolith files', () => {
        assert.file(expectedFiles.monolith);
        assert.fileContent('.yo-rc.json', '"herokuDeployType": "jar"');
        assert.fileContent(`${constants.SERVER_MAIN_RES_DIR}/config/application-heroku.yml`, 'datasource:');
        assert.noFileContent(`${constants.SERVER_MAIN_RES_DIR}/config/application-heroku.yml`, 'mongodb:');
      });
    });

    describe('in the EU', () => {
      let runResult;
      beforeEach(async () => {
        stub.withArgs(`heroku create ${herokuAppName} --region eu`).yields(false, '', '');
        stub.withArgs(`heroku addons:create jawsdb:kitefin --as DATABASE --app ${herokuAppName}`).yields(false, '', '');
        runResult = await helpers
          .create(getGenerator('heroku'))
          .inTmpDir(dir => {
            fse.copySync(getTemplatePath('default/'), dir);
          })
          .withOptions({ skipBuild: true, reproducibleTests: true })
          .withPrompts({
            herokuAppName,
            herokuRegion: 'eu',
            herokuDeployType: 'jar',
            herokuJavaVersion: '11',
            useOkta: false,
          })
          .run();
      });
      it('should match files snapshot', function () {
        expect(runResult.getSnapshot()).toMatchSnapshot();
      });
      it('creates expected monolith files', () => {
        assert.file(expectedFiles.monolith);
      });
    });

    describe('with PostgreSQL', () => {
      let runResult;
      beforeEach(async () => {
        stub.withArgs(`heroku create ${herokuAppName} --region eu`).yields(false, '', '');
        stub.withArgs(`heroku addons:create heroku-postgresql --as DATABASE --app ${herokuAppName}`).yields(false, '', '');
        runResult = await helpers
          .create(getGenerator('heroku'))
          .inTmpDir(dir => {
            fse.copySync(getTemplatePath('default-psql/'), dir);
          })
          .withOptions({ skipBuild: true, reproducibleTests: true })
          .withPrompts({
            herokuAppName,
            herokuRegion: 'eu',
            herokuDeployType: 'jar',
            herokuJavaVersion: '11',
            useOkta: false,
          })
          .run();
      });
      it('should match files snapshot', function () {
        expect(runResult.getSnapshot()).toMatchSnapshot();
      });
      it('creates expected monolith files', () => {
        assert.file(expectedFiles.monolith);
        assert.fileContent(`${constants.SERVER_MAIN_RES_DIR}/config/application-heroku.yml`, 'datasource:');
        assert.noFileContent(`${constants.SERVER_MAIN_RES_DIR}/config/application-heroku.yml`, 'mongodb:');
      });
    });

    describe('with existing app', () => {
      const existingHerokuAppName = 'jhipster-existing';
      let runResult;
      beforeEach(async () => {
        stub
          .withArgs(`heroku apps:info --json ${existingHerokuAppName}`)
          .yields(false, `{"app":{"name":"${existingHerokuAppName}"}, "dynos":[]}`);
        stub.withArgs(`heroku addons:create jawsdb:kitefin --as DATABASE --app ${existingHerokuAppName}`).yields(false, '', '');
        runResult = await helpers
          .create(getGenerator('heroku'))
          .inTmpDir(dir => {
            fse.copySync(getTemplatePath('heroku/'), dir);
          })
          .withOptions({ skipBuild: true, reproducibleTests: true })
          .run();
      });
      it('should match files snapshot', function () {
        expect(runResult.getSnapshot()).toMatchSnapshot();
      });
      it('creates expected monolith files', () => {
        assert.file(expectedFiles.monolith);
        assert.fileContent('.yo-rc.json', `"herokuAppName": "${existingHerokuAppName}"`);
      });
    });

    describe('with elasticsearch', () => {
      let runResult;
      beforeEach(async () => {
        stub.withArgs(`heroku create ${herokuAppName}`).yields(false, '', '');
        stub.withArgs(`heroku addons:create jawsdb:kitefin --as DATABASE --app ${herokuAppName}`).yields(false, '', '');
        stub.withArgs(`heroku addons:create bonsai --as BONSAI --app ${herokuAppName}`).yields(false, '', '');

        runResult = await helpers
          .create(getGenerator('heroku'))
          .inTmpDir(dir => {
            fse.copySync(getTemplatePath('default-elasticsearch/'), dir);
          })
          .withOptions({ skipBuild: true, reproducibleTests: true })
          .withPrompts({
            herokuAppName,
            herokuRegion: 'us',
            herokuDeployType: 'jar',
            herokuJavaVersion: '11',
            useOkta: false,
          })
          .run();
      });
      it('should match files snapshot', function () {
        expect(runResult.getSnapshot()).toMatchSnapshot();
      });
      it('creates expected monolith files', () => {
        assert.file(expectedFiles.monolith);
        assert.fileContent('.yo-rc.json', '"herokuDeployType": "jar"');
        assert.fileContent(`${constants.SERVER_MAIN_RES_DIR}/config/application-heroku.yml`, 'datasource:');
        assert.noFileContent(`${constants.SERVER_MAIN_RES_DIR}/config/application-heroku.yml`, 'mongodb:');
      });
    });
  });
});
