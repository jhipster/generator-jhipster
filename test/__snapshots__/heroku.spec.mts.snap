// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator - Heroku microservice application with JAR deployment should match files snapshot 1`] = `
{
  ".yo-rc.json": {
    "contents": "{
  "generator-jhipster": {
    "applicationType": "microservice",
    "baseName": "jhipster",
    "herokuAppName": "jhipster-test",
    "herokuDeployType": "jar",
    "herokuJavaVersion": "17"
  }
}
",
    "stateCleared": "modified",
  },
  "Procfile": {
    "contents": "web: java $JAVA_OPTS -Xmx256m -jar target/*.jar --spring.profiles.active=prod,heroku
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/application-heroku.yml": {
    "contents": "# ===================================================================
# Spring Boot configuration for the "heroku" profile.
#
# This configuration overrides the application.yml file.
# ===================================================================

# ===================================================================
# Standard Spring Boot properties.
# Full reference is available at:
# http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
# ===================================================================

eureka:
  instance:
    hostname: jhipster-test.herokuapp.com
    non-secure-port: 80
    prefer-ip-address: false

spring:
  datasource:
    type: com.zaxxer.hikari.HikariDataSource
    url: \${JDBC_DATABASE_URL}
    username: \${JDBC_DATABASE_USERNAME}
    password: \${JDBC_DATABASE_PASSWORD}
    hikari:
      maximumPoolSize: 8
server:
  port: \${PORT:8080}
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/bootstrap-heroku.yml": {
    "contents": "# ===================================================================
# Spring Cloud Config bootstrap configuration for the "heroku" profile
# ===================================================================

",
    "stateCleared": "modified",
  },
  "system.properties": {
    "contents": "java.runtime.version=17 ",
    "stateCleared": "modified",
  },
}
`;

exports[`generator - Heroku monolith application in the EU should match files snapshot 1`] = `
{
  ".yo-rc.json": {
    "contents": "{
  "generator-jhipster": {
    "baseName": "jhipster",
    "herokuAppName": "jhipster-test",
    "herokuDeployType": "jar",
    "herokuJavaVersion": "11"
  }
}
",
    "stateCleared": "modified",
  },
  "Procfile": {
    "contents": "web: java $JAVA_OPTS -Xmx256m -jar target/*.jar --spring.profiles.active=prod,heroku
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/application-heroku.yml": {
    "contents": "# ===================================================================
# Spring Boot configuration for the "heroku" profile.
#
# This configuration overrides the application.yml file.
# ===================================================================

# ===================================================================
# Standard Spring Boot properties.
# Full reference is available at:
# http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
# ===================================================================

eureka:
  instance:
    hostname: jhipster-test.herokuapp.com
    non-secure-port: 80
    prefer-ip-address: false

spring:
  datasource:
    type: com.zaxxer.hikari.HikariDataSource
    url: \${JDBC_DATABASE_URL}
    username: \${JDBC_DATABASE_USERNAME}
    password: \${JDBC_DATABASE_PASSWORD}
    hikari:
      maximumPoolSize: 8
server:
  port: \${PORT:8080}
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/bootstrap-heroku.yml": {
    "contents": "# ===================================================================
# Spring Cloud Config bootstrap configuration for the "heroku" profile
# ===================================================================

",
    "stateCleared": "modified",
  },
  "system.properties": {
    "contents": "java.runtime.version=11 ",
    "stateCleared": "modified",
  },
}
`;

exports[`generator - Heroku monolith application in the US should match files snapshot 1`] = `
{
  ".yo-rc.json": {
    "contents": "{
  "generator-jhipster": {
    "baseName": "jhipster",
    "herokuAppName": "jhipster-test",
    "herokuDeployType": "jar",
    "herokuJavaVersion": "11"
  }
}
",
    "stateCleared": "modified",
  },
  "Procfile": {
    "contents": "web: java $JAVA_OPTS -Xmx256m -jar target/*.jar --spring.profiles.active=prod,heroku
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/application-heroku.yml": {
    "contents": "# ===================================================================
# Spring Boot configuration for the "heroku" profile.
#
# This configuration overrides the application.yml file.
# ===================================================================

# ===================================================================
# Standard Spring Boot properties.
# Full reference is available at:
# http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
# ===================================================================

eureka:
  instance:
    hostname: jhipster-test.herokuapp.com
    non-secure-port: 80
    prefer-ip-address: false

spring:
  datasource:
    type: com.zaxxer.hikari.HikariDataSource
    url: \${JDBC_DATABASE_URL}
    username: \${JDBC_DATABASE_USERNAME}
    password: \${JDBC_DATABASE_PASSWORD}
    hikari:
      maximumPoolSize: 8
server:
  port: \${PORT:8080}
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/bootstrap-heroku.yml": {
    "contents": "# ===================================================================
# Spring Cloud Config bootstrap configuration for the "heroku" profile
# ===================================================================

",
    "stateCleared": "modified",
  },
  "system.properties": {
    "contents": "java.runtime.version=11 ",
    "stateCleared": "modified",
  },
}
`;

exports[`generator - Heroku monolith application with Git deployment should match files snapshot 1`] = `
{
  ".yo-rc.json": {
    "contents": "{
  "generator-jhipster": {
    "baseName": "jhipster",
    "herokuAppName": "jhipster-test",
    "herokuDeployType": "git",
    "herokuJavaVersion": "11"
  }
}
",
    "stateCleared": "modified",
  },
  "Procfile": {
    "contents": "web: java $JAVA_OPTS -Xmx256m -jar target/*.jar --spring.profiles.active=prod,heroku
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/application-heroku.yml": {
    "contents": "# ===================================================================
# Spring Boot configuration for the "heroku" profile.
#
# This configuration overrides the application.yml file.
# ===================================================================

# ===================================================================
# Standard Spring Boot properties.
# Full reference is available at:
# http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
# ===================================================================

eureka:
  instance:
    hostname: jhipster-test.herokuapp.com
    non-secure-port: 80
    prefer-ip-address: false

spring:
  datasource:
    type: com.zaxxer.hikari.HikariDataSource
    url: \${JDBC_DATABASE_URL}
    username: \${JDBC_DATABASE_USERNAME}
    password: \${JDBC_DATABASE_PASSWORD}
    hikari:
      maximumPoolSize: 8
server:
  port: \${PORT:8080}
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/bootstrap-heroku.yml": {
    "contents": "# ===================================================================
# Spring Cloud Config bootstrap configuration for the "heroku" profile
# ===================================================================

",
    "stateCleared": "modified",
  },
  "system.properties": {
    "contents": "java.runtime.version=11 ",
    "stateCleared": "modified",
  },
}
`;

exports[`generator - Heroku monolith application with PostgreSQL should match files snapshot 1`] = `
{
  ".yo-rc.json": {
    "contents": "{
  "generator-jhipster": {
    "baseName": "jhipster",
    "herokuAppName": "jhipster-test",
    "herokuDeployType": "jar",
    "herokuJavaVersion": "11"
  }
}
",
    "stateCleared": "modified",
  },
  "Procfile": {
    "contents": "web: java $JAVA_OPTS -Xmx256m -jar target/*.jar --spring.profiles.active=prod,heroku
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/application-heroku.yml": {
    "contents": "# ===================================================================
# Spring Boot configuration for the "heroku" profile.
#
# This configuration overrides the application.yml file.
# ===================================================================

# ===================================================================
# Standard Spring Boot properties.
# Full reference is available at:
# http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
# ===================================================================

eureka:
  instance:
    hostname: jhipster-test.herokuapp.com
    non-secure-port: 80
    prefer-ip-address: false

spring:
  datasource:
    type: com.zaxxer.hikari.HikariDataSource
    url: \${JDBC_DATABASE_URL}
    username: \${JDBC_DATABASE_USERNAME}
    password: \${JDBC_DATABASE_PASSWORD}
    hikari:
      maximumPoolSize: 8
server:
  port: \${PORT:8080}
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/bootstrap-heroku.yml": {
    "contents": "# ===================================================================
# Spring Cloud Config bootstrap configuration for the "heroku" profile
# ===================================================================

",
    "stateCleared": "modified",
  },
  "system.properties": {
    "contents": "java.runtime.version=11 ",
    "stateCleared": "modified",
  },
}
`;

exports[`generator - Heroku monolith application with an unavailable app name should match files snapshot 1`] = `
{
  ".yo-rc.json": {
    "contents": "{
  "generator-jhipster": {
    "baseName": "jhipster",
    "herokuAppName": "jhipster-new-name",
    "herokuDeployType": "jar",
    "herokuJavaVersion": "11"
  }
}
",
    "stateCleared": "modified",
  },
  "Procfile": {
    "contents": "web: java $JAVA_OPTS -Xmx256m -jar target/*.jar --spring.profiles.active=prod,heroku
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/application-heroku.yml": {
    "contents": "# ===================================================================
# Spring Boot configuration for the "heroku" profile.
#
# This configuration overrides the application.yml file.
# ===================================================================

# ===================================================================
# Standard Spring Boot properties.
# Full reference is available at:
# http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
# ===================================================================

eureka:
  instance:
    hostname: jhipster-new-name.herokuapp.com
    non-secure-port: 80
    prefer-ip-address: false

spring:
  datasource:
    type: com.zaxxer.hikari.HikariDataSource
    url: \${JDBC_DATABASE_URL}
    username: \${JDBC_DATABASE_USERNAME}
    password: \${JDBC_DATABASE_PASSWORD}
    hikari:
      maximumPoolSize: 8
server:
  port: \${PORT:8080}
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/bootstrap-heroku.yml": {
    "contents": "# ===================================================================
# Spring Cloud Config bootstrap configuration for the "heroku" profile
# ===================================================================

",
    "stateCleared": "modified",
  },
  "system.properties": {
    "contents": "java.runtime.version=11 ",
    "stateCleared": "modified",
  },
}
`;

exports[`generator - Heroku monolith application with elasticsearch should match files snapshot 1`] = `
{
  ".yo-rc.json": {
    "contents": "{
  "generator-jhipster": {
    "baseName": "jhipster",
    "herokuAppName": "jhipster-test",
    "herokuDeployType": "jar",
    "herokuJavaVersion": "11",
    "searchEngine": "elasticsearch"
  }
}
",
    "stateCleared": "modified",
  },
  "Procfile": {
    "contents": "web: java $JAVA_OPTS -Xmx256m -jar target/*.jar --spring.profiles.active=prod,heroku
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/application-heroku.yml": {
    "contents": "# ===================================================================
# Spring Boot configuration for the "heroku" profile.
#
# This configuration overrides the application.yml file.
# ===================================================================

# ===================================================================
# Standard Spring Boot properties.
# Full reference is available at:
# http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
# ===================================================================

eureka:
  instance:
    hostname: jhipster-test.herokuapp.com
    non-secure-port: 80
    prefer-ip-address: false

spring:
  datasource:
    type: com.zaxxer.hikari.HikariDataSource
    url: \${JDBC_DATABASE_URL}
    username: \${JDBC_DATABASE_USERNAME}
    password: \${JDBC_DATABASE_PASSWORD}
    hikari:
      maximumPoolSize: 8
  elasticsearch:
    uris: \${BONSAI_URL}
server:
  port: \${PORT:8080}
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/bootstrap-heroku.yml": {
    "contents": "# ===================================================================
# Spring Cloud Config bootstrap configuration for the "heroku" profile
# ===================================================================

",
    "stateCleared": "modified",
  },
  "system.properties": {
    "contents": "java.runtime.version=11 ",
    "stateCleared": "modified",
  },
}
`;

exports[`generator - Heroku monolith application with existing app should match files snapshot 1`] = `
{
  ".yo-rc.json": {
    "contents": "{
  "generator-jhipster": {
    "baseName": "jhipster",
    "herokuAppName": "jhipster-existing",
    "herokuDeployType": "git",
    "herokuJavaVersion": "17"
  }
}
",
    "stateCleared": "modified",
  },
  "Procfile": {
    "contents": "web: java $JAVA_OPTS -Xmx256m -jar target/*.jar --spring.profiles.active=prod,heroku
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/application-heroku.yml": {
    "contents": "# ===================================================================
# Spring Boot configuration for the "heroku" profile.
#
# This configuration overrides the application.yml file.
# ===================================================================

# ===================================================================
# Standard Spring Boot properties.
# Full reference is available at:
# http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
# ===================================================================

eureka:
  instance:
    hostname: jhipster-existing.herokuapp.com
    non-secure-port: 80
    prefer-ip-address: false

spring:
  datasource:
    type: com.zaxxer.hikari.HikariDataSource
    url: \${JDBC_DATABASE_URL}
    username: \${JDBC_DATABASE_USERNAME}
    password: \${JDBC_DATABASE_PASSWORD}
    hikari:
      maximumPoolSize: 8
server:
  port: \${PORT:8080}
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/bootstrap-heroku.yml": {
    "contents": "# ===================================================================
# Spring Cloud Config bootstrap configuration for the "heroku" profile
# ===================================================================

",
    "stateCleared": "modified",
  },
  "system.properties": {
    "contents": "java.runtime.version=17 ",
    "stateCleared": "modified",
  },
}
`;
